keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -154,14 +154,25 @@ class ApplicationService(
   ): ArtifactVersionSummary =
     when (artifact.type) {
       deb -> {
-        val appversion = AppVersion.parseName(version)
-        ArtifactVersionSummary(
+        var summary = ArtifactVersionSummary(
           version = version,
           environments = environments,
-          displayName = appversion?.version ?: version.removePrefix("${artifact.name}-"),
-          build = if (appversion != null) BuildMetadata(id = appversion.buildNumber.toInt()) else null,
-          git = if (appversion != null) GitMetadata(commit = appversion.commit) else null
+          displayName = version.removePrefix("${artifact.name}-")
         )
+
+        // attempt to parse helpful info from the appversion.
+        // todo: replace, this is brittle
+        val appversion = AppVersion.parseName(version)
+        if (appversion?.version != null) {
+          summary = summary.copy(displayName = appversion.version)
+        }
+        if (appversion?.buildNumber != null) {
+          summary = summary.copy(build = BuildMetadata(id = appversion.buildNumber.toInt()))
+        }
+        if (appversion?.commit != null) {
+          summary = summary.copy(git = GitMetadata(commit = appversion.commit))
+        }
+        summary
```

Comment:
I'm failing to see how this is different from what you had before, other than setting `displayName` outside this block above. I find this format harder to read than the one with the property names and inline if statements to the left. It's also potentially copying the whole object 3 times over which I know is not a huge deal but I feel if we just let these spread they may become a problem.

Reply:
I believe the current problem is that `appversion.buildNumber` is null for one of the artifacts, so the `appversion.buildNumber.toInt()` call fails.on line 162. In the new implementation, this call is guarded by a null chcek in line 169.

Reply:
Lorin is correct. Again, this code should be replaced in Q2.